civievent-widget.php

```diff
@@ -367,6 +278,183 @@ public function widget( $args, $instance ) {
 		}
 	}
 
+	/**
+	 * Parse the "custom_display" JSON and return list of valid entries
+	 *
+	 * @param array $instance
+	 *   The widget instance.
+	 * @return array
+	 *   The valid "custom_display" entries.
+	 */
+	protected function validCustomDisplayFields ( $instance ) {
+		$fields = $this->getFields();
+		$customDisplay = json_decode( empty( $instance['custom_display'] ) ? '{}' : $instance['custom_display'], true );
+		foreach ( $customDisplay as $name => $fieldAttrs ) {
+			// Make sure only legit fields are sent.
+			if ( empty( $fields[ $name ] ) ) {
+				unset( $customDisplay[ $name ] );
+			}
+		}
+		return $customDisplay;
+	}
+
+	/**
+	 * Parse the "custom_filter" JSON and return list of valid entries
+	 *
+	 * @param array $instance
+	 *   The widget instance.
+	 * @return array
+	 *   The valid "custom_filter" entries.
+	 */
+	protected function validCustomFilterFields ( $instance ) {
+		$fields = $this->getFields();
+		$allCustomDisplayFields = self::getCustomDisplayTitles();
+		$customFilter = json_decode( empty ( $instance['custom_filter'] ) ? '{}' : $instance['custom_filter'], true );
```

Comment:
As in https://github.com/agh1/civievent-widget/pull/7/files#r267834380, we need to wrap this in `html_entity_decode()`.